test/e2e/networking.go

```diff
@@ -71,140 +70,200 @@ var _ = Describe("Networking", func() {
 		}
 	})
 
-	//Now we can proceed with the test.
-	It("should function for intra-pod communication", func() {
-		if testContext.Provider == "vagrant" {
-			By("Skipping test which is broken for vagrant (See https://github.com/GoogleCloudPlatform/kubernetes/issues/3580)")
-			return
-		}
+	//Two parallel array - services[0] corresponds to timeouts[0]
+	services := [...]int{1, 2}         // the first test is equivalent to the original networking.go
+	timeouts := [...]float64{300, 600} // very liberal timeout policies.  Should be more like 100 seconds.
 
-		By(fmt.Sprintf("Creating a service named %q in namespace %q", svcname, f.Namespace.Name))
-		svc, err := f.Client.Services(f.Namespace.Name).Create(&api.Service{
-			ObjectMeta: api.ObjectMeta{
-				Name: svcname,
-				Labels: map[string]string{
-					"name": svcname,
-				},
-			},
-			Spec: api.ServiceSpec{
-				Ports: []api.ServicePort{{
-					Protocol:   "TCP",
-					Port:       8080,
-					TargetPort: util.NewIntOrStringFromInt(8080),
-				}},
-				Selector: map[string]string{
-					"name": svcname,
-				},
+	for i := range services {
+		ii := i
+		It(
+			fmt.Sprintf("should support intrapod communication between all hosts in %v parallel services", services[ii]),
+			func(doneTimeout Done) {
+				RunNetTest(doneTimeout, f, Makeports(services[ii]), "1.4")
 			},
-		})
-		if err != nil {
-			Failf("unable to create test service named [%s] %v", svc.Name, err)
-		}
+			timeouts[ii])
+	}
+})
 
-		// Clean up service
-		defer func() {
-			defer GinkgoRecover()
-			By("Cleaning up the service")
-			if err = f.Client.Services(f.Namespace.Name).Delete(svc.Name); err != nil {
-				Failf("unable to delete svc %v: %v", svc.Name, err)
-			}
-		}()
+//PeerStatus will either fail, pass, or continue polling
+func PollPeerStatus(ch chan int, f *Framework, svc *api.Service) bool {
```

Comment:
your right, in this patch, i just used the same polling as we had before . but using `wait.Poll` or `select`/channels will be a good improvement , maybe for a  later patch.